tests/phpunit/integration/Core/User_Input/User_InputTest.php

```diff
@@ -0,0 +1,200 @@
+		),
+	);
+
+	public function set_up() {
+		parent::set_up();
+		$this->context = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+
+		$this->user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $this->user_id );
+
+		$this->user_input = new User_Input( $this->context );
+		$this->user_input->register();
+	}
+
+	public function test_are_settings_empty() {
+		$data = array(
+			'setting1' => array( 'values' => null ),
+		);
+		$this->assertTrue( $this->user_input->are_settings_empty( $data ) );
+
+		$data = array(
+			'setting1' => array( 'values' => null ),
+			'setting2' => array( 'values' => array( '1', '2', '3' ) ),
+		);
+		$this->assertTrue( $this->user_input->are_settings_empty( $data ) );
+
+		$data = array(
+			'setting1' => array( 'values' => array( 'a', 'b', 'c' ) ),
+			'setting2' => array( 'values' => array( '1', '2', '3' ) ),
+		);
+		$this->assertFalse( $this->user_input->are_settings_empty( $data ) );
+	}
+
+	public function test_get_answers() {
+		// If settings are not set, it returns empty default values.
+		$this->assertEquals(
+			static::$empty_settings,
+			$this->user_input->get_answers()
+		);
+
+		// If settings are partially set, it returns empty default values for unanswered questions.
+		update_user_option(
+			$this->user_id,
+			User_Specific_Questions::OPTION,
+			array(
+				'postFrequency' => array(
+					'values' => array( 'daily' ),
+					'scope'  => 'user',
+				),
+				'goals'         => array(
+					'values' => array( 'goal1', 'goal2' ),
+					'scope'  => 'user',
+				),
+			)
+		);
+
+		$this->assertEquals(
+			array(
+				'purpose'       => array(
+					'values' => array(),
+					'scope'  => 'site',
+				),
+				'postFrequency' => array(
+					'values' => array( 'daily' ),
+					'scope'  => 'user',
+				),
+				'goals'         => array(
+					'values' => array( 'goal1', 'goal2' ),
+					'scope'  => 'user',
+				),
+			),
+			$this->user_input->get_answers()
+		);
+
+		// Returns all set settings as expected.
+		update_option(
+			Site_Specific_Questions::OPTION,
+			array(
+				'purpose' => array(
+					'values'     => array( 'purpose1' ),
+					'scope'      => 'site',
+					'answeredBy' => $this->user_id,
+				),
+			)
+		);
+
+		update_user_option(
+			$this->user_id,
+			User_Specific_Questions::OPTION,
+			array(
+				'postFrequency' => array(
+					'values' => array( 'daily' ),
+					'scope'  => 'user',
+				),
+				'goals'         => array(
+					'values' => array( 'goal1', 'goal2' ),
+					'scope'  => 'user',
+				),
+			)
+		);
+
+		$this->assertEquals(
+			static::$dummy_settings,
+			$this->user_input->get_answers()
+		);
+	}
+
+	public function test_set_answers() {
+		$response = $this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'purpose1' ),
+				'postFrequency' => array( 'daily' ),
+				'goals'         => array( 'goal1', 'goal2' ),
+			)
+		);
+
+		$this->assertEquals(
+			static::$dummy_settings,
+			$response
+		);
+	}
```

Comment:
These are a bit confusing use of the property as there is no clear relationship between what happens in the test and the expected outcome that the value should equal the dummy settings. When using a reference from a property like this, it helps as a reusable value when we aren't overly concerned about what it is. In this case, the test is explicit about the input and ambiguous about the expectation. Another way to think about this, is that as an arbitrary value of valid settings, I would expect to be able to modify that property in any way so long that the values are valid without breaking any of the tests that use it.

Reply:
Removed the properties entirely and inlined the expectations, to be more explicit and readable. Please let me know it this looks good now, thank you!